src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,345 @@
+    {
+        switch (Path.GetExtension(projectPath))
+        {
+            case ".csproj":
+                languageName = LanguageNames.CSharp;
+                return true;
+            default:
+                languageName = null;
+                return false;
+        }
+    }
+
+    private static IReadOnlyList<DocumentInfo> GetDocuments(BuildProjectData buildProjectData, ProjectId projectId) =>
+        (buildProjectData
+            .SourceFiles?
+                .Where(File.Exists)
+                .Select(x => DocumentInfo.Create(
+                    DocumentId.CreateNewId(projectId),
+                    Path.GetFileName(x),
+                    loader: TextLoader.From(
+                        TextAndVersion.Create(
+                        SourceText.From(File.ReadAllText(x), Encoding.Unicode), VersionStamp.Create())),
+                    filePath: x))
+            ?? Array.Empty<DocumentInfo>()).ToList();
+
+    private static IReadOnlyList<ProjectReference> GetExistingProjectReferences(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace) =>
+        (buildProjectData.ProjectReferences
+            .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath.Equals(x, StringComparison.OrdinalIgnoreCase)))
+            .Where(x => x != null)
+            .Select(x => new ProjectReference(x.Id))
+        ?? Array.Empty<ProjectReference>()).ToList();
+
+    private static IEnumerable<MetadataReference> GetMetadataReferences(BuildProjectData buildProjectData) =>
+        buildProjectData
+            .References?.Where(File.Exists)
+            .Select(x => MetadataReference.CreateFromFile(x))
+        ?? (IEnumerable<MetadataReference>)Array.Empty<MetadataReference>();
+
+
+    private static IReadOnlyList<AnalyzerReference> GetAnalyzerReferences(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace)
+    {
+        IAnalyzerAssemblyLoader loader = workspace.Services.GetRequiredService<IAnalyzerService>().GetLoader();
+
+        return (buildProjectData
+                .AnalyzerReferences?
+                    .Where(File.Exists)
+                    .Select(x => new AnalyzerFileReference(x, loader))
+                ?? (IEnumerable<AnalyzerReference>)Array.Empty<AnalyzerReference>()).ToList();
+    }
+
+    private static ParseOptions CreateParseOptions(BuildProjectData buildProjectData, string languageName)
+    {
+        // language-specific code is in local functions, to prevent assembly loading failures when assembly for the other language is not available
+        if (languageName == LanguageNames.CSharp)
+        {
+            return CreateCSharpParseOptions(buildProjectData);
+        }
+
+        return null;
+    }
+
+    private static CSharpParseOptions CreateCSharpParseOptions(BuildProjectData buildProjectData)
+    {
+        CSharpParseOptions parseOptions = new CSharpParseOptions();
+
+        // Add any constants
```

Comment:
Feel free to remove comments like this. They don't add useful information.

Reply:
removed most of the comments since the look redundant.